boot/systems.go

```diff
@@ -342,3 +343,93 @@ func InspectTryRecoverySystemOutcome(dev Device) (outcome TryRecoverySystemOutco
 
 	return outcome, trySystem, nil
 }
+
+// MarkRecoverySystemGood marks the provided recovery system as a good one, and
+// promotes it to the list of good recovery systems in modeenv. If the system
+// uses encryption, the keys will updated state. If resealing fails, an attempt
+// to restore the previous state is made
+func MarkRecoverySystemGood(dev Device, systemLabel string) (err error) {
+	if !dev.HasModeenv() {
+		return fmt.Errorf("internal error: recovery systems can only be used on UC20")
+	}
+
+	m, err := loadModeenv()
+	if err != nil {
+		return err
+	}
+	if !strutil.ListContains(m.CurrentRecoverySystems, systemLabel) {
```

Comment:
does this check make sense,  afair the code we added in devicemgr.go wil clear the tried label out of this list?  so I'm entirely sure how this is expected to be used, will something add it back first?

Reply:
No, you're right. Also we rewrite the modeenv using AtomicFile so in theory we should have both entries or none. This is how it's used, at least in the non-remodel scenario.

Reply:
Hm, was this addressed and the conversation is stale?

Reply:
it was addressed, the original code in conditional stanza when I made the comment was returning an error in this case